tests/openvino/tools/calibrate.py

```diff
@@ -0,0 +1,267 @@
+
+   
+def map_target_device(target_device):
+    target_device = target_device.upper()
+    if target_device not in [t.value for t in nncf.TargetDevice]:
+        raise ValueError(f'{target_device} target device is not supported')
+    return {'target_device': nncf.TargetDevice(target_device)}
+
+
+def map_model_type(model_type):
+    if model_type is None:
+        return None
+    
+    model_type = model_type.lower()
+    if model_type not in [m.value for m in nncf.ModelType]:
+        raise ValueError(f'{model_type} model type is not supported')
+    return {'model_type': nncf.ModelType(model_type)}
+
+
+def map_ignored_scope(ignored):
+    if ignored is None:
+        return None
+    
+    if ignored.get('skip_model') is not None:
+        raise ValueError(f'skip_model attribute in the ignored tag is not '
+                         f'supported')
+    
+    operations  = ignored.get('operations')
+    if operations is not None:
+        for op in operations:
+            if op.get('attributes') is not None:
+                raise ValueError(f'Attributes in the ignored operations '
+                                 f'are not supported')
+    return {'ignored_scope': nncf.IgnoredScope(names=ignored.get('scope'), 
+                                               types=ignored.get('types'))}
+
+
+def map_preset(preset):
+    preset = preset.lower()
+    if preset not in [p.value for p in nncf.QuantizationPreset]:
+        raise ValueError(f'{preset} preset is not supported')
+    return {'preset': nncf.QuantizationPreset(preset)}
+
+
+def map_quantization_parameters(pot_parameters):    
+    def _map_parameters(parameters, supported_parameters, 
+                        default_parameters, ignored_parameters):
+        result = {}
+        for name in parameters:
+            if (name in ignored_parameters or 
+                (name in default_parameters and 
+                 parameters[name] == default_parameters[name])):
+                continue
+              
+            if name in supported_parameters:
+                kwarg = parameters_mapping[name](parameters[name])
+                if kwarg is not None:
+                    result.update(kwarg)
+            elif parameters[name] is not None:
+                _map_parameters(parameters[name], {}, {}, [])
+            else:
+                raise ValueError(f'{name} parameter is not supported')
+        return result
+
+    parameters_mapping = {
+        'target_device': map_target_device,
+        'model_type': map_model_type,
+        'ignored': map_ignored_scope,
+        'preset': map_preset,
+        'stat_subset_size': lambda x: {'subset_size': x},
+        'use_fast_bias': lambda x: {'fast_bias_correction': x}
+    }
+    
+    default_parameters = {
+      'use_layerwise_tuning': False
+    }
+
+    ignored_parameters = [
+        'dump_intermediate_model',
+        'inplace_statistics',
+        'num_samples_for_tuning'
```

Comment:
There is also a `tuning_iterations` parameter for `LayerwiseTuning`. Also about the `sparsity_level` parameter (that might be useless now).

Reply:
done